example_envs/bookinfo/leafnode.rs

```diff
@@ -32,25 +32,41 @@ impl SimElement for LeafNode {
                 let deq = self.core_node.dequeue(tick);
                 rpc = deq.unwrap();
             } else {
-                // no rpc in the queue, we only react so nothing to do
+                // no rpc in the queue, we only forward so nothing to do
                 continue;
             }
-            // "respond to every node that has sent a request"
+            // forward requests/responses from productpage or reviews
             if rpc.headers.contains_key("src") {
-                let target = rpc.headers["src"].to_string();
-                rpc.headers
-                    .insert("direction".to_string(), "response".to_string());
+                let dest = self.choose_destination(&rpc);
+                rpc.headers.insert("dest".to_string(), dest.clone());
                 rpc.headers
                     .insert("src".to_string(), self.core_node.id.to_string());
-                ret.push((rpc, target));
+                rpc.headers
```

Comment:
Should only be pushed when we have a plugin.

Reply:
Why?  It doesn't hurt to have extra ingress/egress information when we don't have a plugin.

Reply:
Well, it does not serve a purpose outside of that and the location will also be incorrect at the receiving node.

Reply:
Okay, that makes sense.